tests/schema_tests2.rs

```diff
@@ -212,6 +216,174 @@ pub mod tests {
 
         let db = Db::init(db_path).unwrap();
         assert!(db.table3.get_all().unwrap().is_empty());
-        fs::remove_dir_all(db_path).unwrap_or_else(|_| {});
+        fs::remove_dir_all(db_path).unwrap();
+    }
+
+    #[test]
+    fn test_log_compaction_1() {
+        let db_path = &test_db();
+
+        fs::remove_dir_all(db_path).unwrap();
+        let db = Db::init(db_path).unwrap();
+
+        db.table1.insert(Test {}, "test".to_string()).unwrap();
+        db.table1.insert(Test {}, "test".to_string()).unwrap();
+        db.table1.insert(Test {}, "test".to_string()).unwrap();
+        db.table1.insert(Test {}, "test".to_string()).unwrap();
+
+        let size_before = File::open(&format!("{}/{}", db_path, SCHEMA_NAME))
+            .unwrap()
+            .metadata()
+            .unwrap()
+            .len();
+        db.compact_log().unwrap();
+        let size_after = File::open(&format!("{}/{}", db_path, SCHEMA_NAME))
+            .unwrap()
+            .metadata()
+            .unwrap()
+            .len();
+
+        assert_eq!(size_before, 112);
+        assert_eq!(size_after, 36);
+
+        fs::remove_dir_all(db_path).unwrap();
+    }
+
+    #[test]
+    fn test_log_compaction_2() {
+        let db_path = &test_db();
+
+        fs::remove_dir_all(db_path).unwrap();
+        let db = Db::init(db_path).unwrap();
+
+        db.table2.insert(Test {}, 123).unwrap();
+        db.table2.insert(Test {}, 124).unwrap();
+        db.table2.insert(Test {}, 125).unwrap();
+        db.table2.insert(Test {}, 126).unwrap();
+
+        db.table3.insert("a".to_string(), vec![3, 2, 1]).unwrap();
+        db.table3.insert("a".to_string(), vec![1, 2, 3]).unwrap();
+        db.table3.insert("b".to_string(), vec![1]).unwrap();
+        db.table3.insert("b".to_string(), vec![1, 3]).unwrap();
+
+        let size_before = File::open(&format!("{}/{}", db_path, SCHEMA_NAME))
+            .unwrap()
+            .metadata()
+            .unwrap()
+            .len();
+        db.compact_log().unwrap();
+        let size_after = File::open(&format!("{}/{}", db_path, SCHEMA_NAME))
+            .unwrap()
+            .metadata()
+            .unwrap()
+            .len();
+
+        assert_eq!(size_before, 269);
+        assert_eq!(size_after, 95);
```

Comment:
how do you know these are the right numbers

Reply:
you should literally get the values out of the database and make sure they are what you put in